mlflow/pyfunc/__init__.py

```diff
@@ -3109,11 +3115,30 @@ def predict(self, context, model_input: List[str], params=None) -> List[str]:
             else:
                 # TODO: validate input example against signature
                 update_signature_for_type_hint_from_example(input_example, mlflow_model.signature)
-
-    if metadata is not None:
-        mlflow_model.metadata = metadata
-    if saved_example is None:
-        saved_example = _save_example(mlflow_model, input_example, path, example_no_conversion)
+        else:
+            if saved_example is None:
+                color_warning(
+                    message="An input example was not provided when logging the model. To ensure "
```

Comment:
nit: we may want to change indents to make it more readable.
```python
                color_warning(
                    message="An input example was not provided when logging the model. "
                            "To ensure the model signature functions correctly,"
                            "specify the `input_example` parameter. See"
                            "https://mlflow.org/docs/latest/model/signatures.html#model-input-example "
                            "for more details about the benefits of using input_example."
```

Reply:
It's actually autoformat 😅